sw/device/tests/i2c_host_power_monitor_test.c

```diff
@@ -0,0 +1,173 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+#include <assert.h>
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_i2c.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/irq.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/lib/testing/i2c_testutils.h"
+#include "sw/device/lib/testing/rv_core_ibex_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "i2c_regs.h"  // Generated.
+
+static_assert(__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__,
+              "This test assumes the target platform is little endian.");
+
+OTTF_DEFINE_TEST_CONFIG();
+
+enum {
+  kDeviceAddr = 0x10,
+  kManufacturerId = 0xFE,
+  kProductId = 0xFD,
+  kAccumConfig = 0x25,
+  kCtrAct = 0x21,
+  kVSensenAvg = 0x13,
+  kVPowerN = 0x17,
+  kOcLimitN = 0x30,
+};
+/**
+ * Declared volatile because it is referenced in the main program flow as well
+ * as the ISR.
+ */
+// Hold the test result.
+static volatile status_t test_result;
```

Comment:
nit: this no longer needs to be volatile.  It could also be a local in `main` rather than a global.

Reply:
Done